### PR TITLE
Zero gravity movement will now ignore injury/hunger/clothing/fat/body temperature slowdowns if you are using a jetpack.

### LINKED ISSUE #8360: Wounded Jetpacking
I noticed that I was slowed down in space after being wounded by carp even though I was just strolling around with a jetpack, not walking on girders. Could've been just space lag but sure seemed to slow down. Wound was mainly on leg if that matters.

### PR DESCRIPTION
**Pull request by xxalpha**
Fixes #8360

### CODE REVIEW

**Comment by JJRcop** [comment]
> Adds fun.
:-1:

**Comment by WJohn** [comment]
:+1: x :100:

**Comment by xxalpha** [comment]
Why not @JJRcop ?

**Comment by Fox-McCloud** [comment]
:+1: 

Definitely agree with this; it's pretty unrealistic to be slowed if you're in a no grav environment.

**Comment by JJRcop** [comment]
@xxalpha Just forwarding the stereotype. Jokes aside: :thumbsup:

**Comment by optimumtact** [comment]
The new code looks less asthetic than my version :-1:

**Comment by RemieRichards** [comment]
Have you actually tested this? does it work?

Because looking at it it shouldn't make a difference...

Old version:
Add all the slowdowns, if somebody has IGNORESLOWDOWN, set the variable back to 0 then handle speedups

New version:
Only add slowdowns if somebody doesn't have IGNORESLOWDOWN, handle speedups.

It's the same outcome.

File: code/modules/mob/living/carbon/human/species.dm [hunk 1]
```diff
@@ -657,40 +657,39 @@
...
-		mspeed += H.back.slowdown
+				var/hungry = (500 - H.nutrition) / 5	//So overeat would be 100 and default level would be 80
+				if(hungry >= 70)
+					mspeed += hungry / 50
 
-	if((H.disabilities & FAT) && grav)
-		mspeed += 1.5
-	if(H.bodytemperature < 283.222)
-		mspeed += (283.222 - H.bodytemperature) / 10 * (grav+0.5)
+				if(H.wear_suit && grav)
```
Comment (optimumtact) [inline code/modules/mob/living/carbon/human/species.dm]: no longer need to check grav individually here

**Comment by xxalpha** [comment]
I am very aware of how ugly it turned out to be, I am open to suggestions to make it easier to read.

**Comment by Steelpoint** [comment]
From a gameplay perspective :+1:

**Comment by optimumtact** [comment]
http://pastie.org/10029450

**Comment by Dorsisdwarf** [comment]
Praise the funbus!

**Comment by MrPerson** [comment]
I don't understand what positive gameplay changes you're trying to create here. I understand and agree with having jetpacking not apply slowdowns, that's fine, but think about what behavior this will encourage. Instead of going through the station, people will go around the station because it'll be faster AND safer. Going through the station is a better option from a design standpoint because it allows interaction with the rest of the crew, which is literally the only thing that makes the game fun to begin with. So in other words, this PR encourages people to not have fun. We should be encouraging people to go through the station, which means much slower movement when going around.

Removing some of these slowdowns from space completely eliminates the entire purpose of them existing. Food applies a slowdown so you go back to the station to remove it. Making it not apply in space literally encourages you to stay IN SPACE AWAY FROM THE STATION CONSTANTLY, which is really fucking stupid! WIthout hardsuit slowdown, https://tgstation13.org/phpBB/viewtopic.php?f=5&t=3125 is a complete no-go. Hardsuit slowdown is a thing on the station so people don't just wear them constantly. See the first paragraph for why this should also apply in space. Cold slowdown is irrelevant in this case, fat slowdown is irrelevant in general, boots is so you don't wear magboots on constantly (which imo is stupid but w/e). I don't even know what back slowdown is supposed to be, so I can't comment on it.

Please don't encourage people to be boring loner powergamers. Instead try to get people back on the station so they'll interact with the crew and actually have fun.

**Comment by xxalpha** [comment]
Now you only won't be slowed down if you are using a jetpack.

**Comment by Miauw62** [comment]
I agree with MrPerson, but I also agree that the delay should be removed
when using a jetpack. There's far less jetpacks on the station than there
are spacesuits, and usually they're not *that* useful. This could help.

**Comment by optimumtact** [comment]
Happy with this now :+1:

**Comment by Fox-McCloud** [comment]
Because, you know, there's so much to do in space as a loner anyway---oh wait.

In theory, it sounds nice, but the fact you move faster is more than negated by the fact that your points of entry are limited and you have to take a very long path around the station to get from point A to point B. 

What next? Bay style lethal disposals because it "encourages powergamers to be loners and avoid the station"?

All this is going to accomplish is for powergamers to seek out hyperzine (or any of the other movement boosters) and continue doing the same exact thing.

**Comment by MrPerson** [comment]
Having not a lot to do in space is all the more reason to keep people on the station. Or add shit to do in space, I suppose, but I'd rather add more to the station.

If someone is taking hyperzine and using it to go around the station, good for them. They get rewarded for their planning. Abuse should be reigned in by goofball's chemistry stuff.

**Comment by Fox-McCloud** [comment]
@MrPerson 

So getting hyperzine or some other chems is an investment, but getting a firesuit+silver sulfa or an EVA suit is not?

Only people that have easy access to space-faring gear is engineering

**Comment by Lumipharon** [comment]
There is 3 jetpacks on station. As you seriously worried about people taking one of these and then floating around the station FOR NO REASON, not interacting with people.

**Comment by dannno** [comment]
@MrPerson 
smdh
this is so not a big deal at all
you are overthinking this *hard*

**Comment by WJohn** [comment]
Yeah seriously the number of people who will consider taking the time to get a space suit, put it on, go to one end of the station, circle around (and throwing away items as they slip here and there), then making their own entrance AS OPPOSED to just walking through the hallways in ten seconds... it's not an issue.

**Comment by Steelpoint** [comment]
Yep, its much quicker to just walk inside the station to your destination rather than take the effort to get EVA gear and go spacewalking.

**Comment by dannno** [comment]
i am utterly shocked at the length of that post
what the fuck mrperson

**Comment by optimumtact** [comment]
god forbid someone puts some effort into explaining their opinions right dannno? Keep it on the short and simple meme level where you can understand it

**Comment by Dorsisdwarf** [comment]
I feel midly obligated to point out that slowspace was the single most reviled and hated change that was ever made on this server, worse than runspeed, worse than stun rebalance, worse than taser nerfs, worse than pokecult removal.

It's illogical, it's bad game design, it's annoying, it adds nothing whatsoever to the game, and for good measure it's completely unrealistic (if you're the kind who gets het up at all that kind of stuff)

**Comment by Jalleo** [comment]
The only times i would take a spacewalk is to get from engineering to escape. It makes logical sense if you do have a spacesuit. Anything else is normally slower. Literally only hull damage repair to loop round the station. That never happens anymore.

**Comment by xxalpha** [comment]
As was pointed out to me, the original change would undermine some things and bring back other bad things. This PR will now fix the Issue only. I am thinking of another thing to help Engineers deal with slowspace.

**Comment by Steelpoint** [comment]
So what exactly are the negative reasoning's behind not allowing people in space to move faster? The reasoning's presented seem flimsy at best as its been well stated how its absurd to think that somehow that by making space faster that people who want to get from point A to B will suddenly decided to detour through space, as well as state that doing this will be faster than just walking from point A to B instead of the massive detour.

I can safely say that the vast majority of the crew will never leave the station z-level, there's little content in space to be worthwhile exploring to begin with.

If I want to be a "powergaming loner" I can just not interact with anyone or stick to maintenance, being in space has nothing to do with that.

**Comment by xxalpha** [comment]
Makes hunger irrelevant, also remember the magboots rampage?

**Comment by Lumipharon** [comment]
Makes hunger irrelevent when DRIFTING THROUGH SPACE under the power of a jetpack. What is the issue?

**Comment by xxalpha** [comment]
Calm the fuck down and read the PR.

**Comment by dannno** [comment]
@optimumtact don't you think you're exploiting what I said a little
yeah this was obv. an attack on all well-thought out explanations man i sure hate those

**Comment by Dorsisdwarf** [comment]
I retract my statement about hail the fun.
Fun removed.

**Comment by Jalleo** [comment]
I do know fast space made it impossible to fight against people in space pre nerf. What is it like with this change?

**Comment by xxalpha** [comment]
Zero gravity movement suffers no slowdowns if you have a working jetpack equipped, meaning you don't get slowed down because there is no gravity (there's a slowdown for that) or any slowdowns caused by injuries, hunger, etc. Basically standard speed without slowdown.

**Comment by Jalleo** [comment]
So definitly same speed as someone on station without armor?
If so this is kinda balanced for carp but for sec it may become a bit unbalanced with station to space combat due to armor n food. This could become a prime powergamer tactic. How about making standard ones a bit slower and keep the syndicate ones at this to make a good counter balance?

**Comment by Lumipharon** [comment]
if you're trying to fight someone when you're at an objective advantage, it's your own fault.
Someone in space with a jetpack can't do shit unless you actually try fight him IN SPACE.

**Comment by MrPerson** [comment]
You get an advantage but, in theory, we could balance it by requiring the jetpacking guy come in to refill the tanks.

**Pull request merged**